src/cmd-compress

```diff
@@ -41,6 +44,24 @@ print(f"Targeting build: {build}")
 imgs_to_skip = ["iso", "vmware", "initramfs", "kernel"]
 
 
+def get_cpu_param(param):
+    with open(f'/sys/fs/cgroup/cpu/cpu.{param}') as f:
+        return int(f.read().strip())
+
+
+# XXX: should dedupe this with logic in virt-install and put in the shared lib
+def xz_threads():
+    with open("/proc/1/cgroup") as f:
+        in_k8s = re.search(r'.*kubepods.*', f.read())
+    if in_k8s:
+        quota = get_cpu_param('cfs_quota_us')
+        # are k8s limits enforced?
```

Comment:
Did you invent this code or take it from somewhere else?

Reply:
I re-read https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/ and read https://medium.com/@betz.mark/understanding-resource-limits-in-kubernetes-cpu-time-9eff74d3161b.

It's also mostly the same thing the JDK HotSpot code does too though: https://hg.openjdk.java.net/jdk/jdk/file/7b671e6b0d5b/src/hotspot/os/linux/osContainer_linux.cpp#l595. Which, hmm yeah, should probably check that the period is `> 0`.

Reply:
OK, threw a link to that and added a check for the period.

I also found https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpu really useful. Essentially though, I'm trying to avoid xz scheduling e.g. 20 threads because it sees 20 CPUs when really it'd be throttled at e.g. `200m`.